FEMO/Space_board/GameScene.swift

```diff
@@ -102,12 +107,21 @@ class GameScene: SKScene {
     }
     
     override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
-        if let touch = touches.first {
-            let location = touch.previousLocation(in: self)
-            let node = self.nodes(at: location).first
+        let touch = touches.first
+        
+        if let location = touch?.location(in: self) {
+            let node = self.nodes(at: location)
             
-            if node?.name == "nextTileButton" {
+            if node.first?.name == "nextTileButton" {
                 playTurn()
+            } else if node.first?.name == "Menu_button" {
```

Comment:
Looks great! We just need to add a line here to save the state of the board when transitioning. Same thing goes for the info button.

Reply:
completed